lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -123,42 +124,76 @@ private void Launch(IJobSubmission jobSubmission, string driverFolderPath)
 
             // TODO: Remove this when we have a generalized way to pass config to java
             var paramInjector = TangFactory.GetTang().NewInjector(jobSubmission.DriverConfigurations.ToArray());
-                
-            var avroJobSubmissionParameters = new AvroJobSubmissionParameters
+
+            var submissionJobArgsFilePath = SerializeJobFile(jobSubmission, driverFolderPath);
+            var submissionAppArgsFilePath = SerializeAppFile(jobSubmission, paramInjector, driverFolderPath);
+
+            // Submit the driver
+            _javaClientLauncher.Launch(JavaClassName, submissionJobArgsFilePath, submissionAppArgsFilePath);
+            Logger.Log(Level.Info, "Submitted the Driver for execution." + jobSubmission.JobIdentifier);
+        }
+
+        private string SerializeAppFile(IJobSubmission jobSubmission, IInjector paramInjector, string driverFolderPath)
```

Comment:
I agree that it looks kind of weird and we can definitely abstract it out into a separate concrete class, but I'm not sure of the utility of abstracting it into a hierarchy of interfaces, mainly for the following reasons:

1. Only YARN client needs it (so far).
2. The two YARN clients will be merged anyway.
3. The arguments and return types are different in both instances.

Given what we have so far, I think we should table the work to create an interface hierarchy until we can generalize better. What do you think?